include/envoy/secret/secret_manager.h

```diff
@@ -70,11 +70,13 @@ class SecretManager {
    * @param config_name a name that uniquely refers to the SDS config source.
    * @param secret_provider_context context that provides components for creating and initializing
    * secret provider.
+   * @param extra_key that is needed to find a dynamic secret provider.
```

Comment:
rename extra_key to extra_sharing_key.

extra_sharing_key is to control how secret providers can be shared. Usually they are shared if sds_config is the same. But for CombinedValidationContext case, they should not be share if their default_validation_context is not the same.  the extra_sharing_key will pass in default_validation_context

Reply:
Done.